src/Plugin/Visualisation/Style/AxisChart.php

```diff
@@ -742,30 +742,39 @@ public function build() {
     $build = [];
 
     foreach ($this->getSourceRecords() as $group_key => $group_records) {
-      $build[$group_key]['chart'] = [
+      $group_id = strtolower(Html::cleanCssIdentifier($group_key));
+
+      $build[$group_id] = [
```

Comment:
@WilsonWu2100 this was the main issue for the download buttons, each split chart didn't have its own wrapper, meaning jQuery just grabbed the first chart

Reply:
@jez500 I've tested this, it's working perfectly. Thank you.